Backend/Frontend Project -Jasmin

JasminHed opened the pull request:
Render: https://js-project-api-x10r.onrender.com
Netlify: https://apifrontback.netlify.app/

JasminHed committed:
added api endpoint all data + added enpoint messages for all messages + added id + find method for 1 message data

JasminHed committed:
changed the endpoint function to get alla messages + changed filter to hearts and not colors

JasminHed committed:
added frontend folder + code in app.jsx

JasminHed committed:
console log for debug backend

JasminHed committed:
updated data.json import

JasminHed committed:
console log

JasminHed committed:
commented out code

JasminHed committed:
added console log

JasminHed committed:
added mongourl, schema + model and changed to async, try, catch function

JasminHed committed:
added new deployed url

JasminHed committed:
updated the cors

JasminHed committed:
changed cors back

JasminHed committed:
added form for adding happy thoughts

JasminHed committed:
added edit + delete

JasminHed committed:
updated button styling

JasminHed committed:
added file messagecard, styled buttons + added handle likes function + POST + like button

JasminHed committed:
added a POST function for likes on messages

JasminHed committed:
changed width on mobile

JasminHed committed:
updated styling for mobile

JasminHed committed:
added a modelthoughfile in backend

JasminHed committed:
added the user schema+model in seperate file

JasminHed committed:
added register route in server.js

JasminHed committed:
changed user model, username to name and added token as in van video

JasminHed committed:
added code snippet/middleware that checks authorization w accesstoken

JasminHed committed:
added the line authenticateuser to each route for saftey

JasminHed committed:
added 2 new errro messages for post user endpoint, email adress exist and username exist

JasminHed committed:
login and register frontend

JasminHed committed:
added Authorization, accesstoken in frontend delete+edit+postmessages

JasminHed committed:
added register+login in app.jsx return

JasminHed committed:
added origin netlify under cors in server.js

JasminHed committed:
console log in register

JasminHed committed:
added success message registration+login

JasminHed committed:
hard brackets around netlify origin in cors

JasminHed committed:
commented out user required in thought model

JasminHed committed:
added localhost in cors

JasminHed committed:
updated logincode with comments

JasminHed committed:
added console log for user and login

JasminHed committed:
added styling to log in form + button, toggle

JasminHed committed:
frontend login and registration in one w.props

JasminHed committed:
frontend mobil styling

JasminHed committed:
empty input in log in, added close button on form, validate password in frontend

JasminHed committed:
close button, password controll, form styling

JasminHed committed:
automatic update of logged out user and logged in user through moving the login handle to app.jsx, global. page still reloads-will fix

JasminHed committed:
updated comments

JasminHed committed:
comment for password crypto in server.js +form button hover + input filed login+register width+changed h1 in globalstyle

JasminHed committed:
comment cryp password for show

JasminHed committed:
added white border upon hover all buttons, settimeout 3 sec for success messages + empty input field

JasminHed committed:
removed setloadingtrue, page no longer reloads

JasminHed committed:
changed colors for better accessability

JasminHed committed:
test color

JasminHed committed:
new color buttons

JasminHed committed:
new color for get button

JasminHed committed:
h1 more margin top + other color for loginform for better visibility

christina-baldwin commented on backend/server.js:
```diff
@@ -0,0 +1,277 @@
+dotenv.config()
+
+//atlas database from env.file or go to local (?)
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
```
If you have your mongodb url in your env file you might not need the alternative local host? As i think its better to just take it from your env file, but i could be wrong

JasminHed replied:
Smart, I will comment it out!

christina-baldwin commented on backend/server.js:
```diff
@@ -0,0 +1,277 @@
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
+mongoose.connect(mongoUrl) 
+
+const port = process.env.PORT || 8080;
```
Same here as the above comment, I think

christina-baldwin commented on backend/server.js:
```diff
@@ -0,0 +1,277 @@
+
+// Add middlewares to enable cors and json body parsing
+app.use(cors({
+  origin: ["https://apifrontback.netlify.app", "http://localhost:5173" ] //allow netlify
```
Genuinely curious because I don't know, what does this do?

JasminHed replied:
Great question, so as I understand it this middleware allows request from my frontend, netlify and localhost. I am not sure if this is the way to go but as I was working woth connecting I added both to be able to test the connection.

christina-baldwin commented on frontend/public/vite.svg:
Unless you are using these svg's I don't think you need to keep them in your folder as they'll just take up space in your repo (again, personal opinion)

JasminHed replied:
Great observation, just removed :)

christina-baldwin reviewed (commented):
I remember your presentation of this too, and everything looks good and works well, I have no major comments really, great job! :D

JasminHed committed:
removed svg

JasminHed commented:
Thank you for looking over my code Christina, always great to have input from someone who knows more than yourself 🥇

JasminHed committed:
updated readme file

JasminHed committed:
page regions

JennieDalgren reviewed (approved):
Good job! Your project is approved. There are some UX details to think about if you want to improve. A user can right now delete all thoughts and edit even other user's thoughts. 
You could prevent the form from being submitted empty. Right now you get an error message just saying "Could not post happy thought" when submitting an empty thought

Det är lite svårt att läsa vad det står på registreringsformuläret 
<img width="1170" height="835" alt="image" src="https://github.com/user-attachments/assets/af5ce68e-ea63-4c03-ba94-9b43ca870f61" />

The code is well strcutured and easy to follow!

JasminHed committed:
updated backend with pst connected to id so that only user can delete thoght

JasminHed committed:
updated backend  edit to connect to user

JasminHed committed:
updated frontend to check for user, only show edit and delete when its the users

JasminHed committed:
updated messagecard to match object+string

JasminHed committed:
edit + delete only show on post that belong to log
ged in user + checks if form is empty, rendersa message please add happy thought

JasminHed commented:
Thank you Jennie! The delete and edit button is now only rendered on the post of the logged in user, which means only they can delete and edit their post and no one elses. This saved me for the final project as well!! Also added a better color for the form :) and made sure that when the field is empty on happy thought you get a message prompting you to add a happy thought before continuing.
